lib/manager/gradle-wrapper/artifacts.ts

```diff
@@ -0,0 +1,82 @@
+import Git from 'simple-git/promise';
+import { resolve } from 'path';
+import * as fs from 'fs-extra';
+import { logger } from '../../logger';
+import { UpdateArtifact, UpdateArtifactsResult } from '../common';
+import { exec, ExecOptions } from '../../util/exec';
+import { readLocalFile } from '../../util/fs';
+import { platform } from '../../platform';
+import { VERSION_REGEX } from './search';
+import { gradleWrapperFileName, prepareGradleCommand } from '../gradle/index';
+
+async function addIfUpdated(
+  status: Git.StatusResult,
+  filePath: string
+): Promise<UpdateArtifactsResult | null> {
+  if (status.modified.includes(filePath)) {
+    return {
+      artifactError: null,
+      file: {
+        name: filePath,
+        contents: await readLocalFile(filePath),
+      },
+    };
+  }
+  return null;
+}
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  try {
+    const projectDir = resolve(packageFileName, './../../../');
+    logger.debug(updatedDeps, 'gradle-wrapper.updateArtifacts()');
+    const gradlewPath = resolve(
+      projectDir,
+      `./${gradleWrapperFileName(config)}`
+    );
+    const version = VERSION_REGEX.exec(newPackageFileContent).groups.version;
+    prepareGradleCommand(
+      gradleWrapperFileName(config),
+      projectDir,
+      await fs.stat(gradlewPath).catch(() => null),
+      null
+    );
+    const cmd = `${gradlewPath} wrapper --gradle-version ${version} --project-dir ${projectDir}`;
+    logger.debug(`Updating gradle wrapper: "${cmd}"`);
+    const execOptions: ExecOptions = {
+      docker: {
+        image: 'renovate/gradle',
+      },
+    };
+    try {
+      await exec(cmd, execOptions);
+    } catch (err) /* istanbul ignore next */ {
+      logger.warn(
+        { err },
+        'Error executing gradle wrapper update command. It can be not a critical one though.'
```

Comment:
Is it better to warn, or to treat this as an "artifacts error" like other managers?

Reply:
We should only warn, because sometimes the execution command exits with a non-zero exit code, although everything was ok. (This happens in windows with `gradlew.bat`).